service/create/service.go

```diff
@@ -1294,34 +1329,61 @@ func (s *Service) deleteFunc(obj interface{}) {
 		s.logger.Log("error", fmt.Sprintf("%#v", err))
 	}
 
-	// Delete gateway.
-	var gateway resources.ResourceWithID
-	gateway = &awsresources.Gateway{
+	if key.HasClusterVersion(cluster) {
+		// Delete NAT gateway.
+		var natGateway resources.ResourceWithID
```

Comment:
nitpick, do we need to declare the variable separately, why not using `:=` on assignment?

Reply:
Good point, removed.